### PR TITLE
Upgraded gems and fixed consequences

### LINKED ISSUE #246: Upgrade oneview-sdk gem to version 5.0.0
### Scenario/Intent
Update oneview-sdk gem version to 5.0.0 that will provide support to HPE OneView API500 resources.

Update also the other gems if possible/applicable.

### Environment Details
 - **oneview Cookbook Version:** 2.3.0
 - **OneView Version:** 3.10

### PR DESCRIPTION
**Pull request by tmiotto**
### Description
Updates the gems, includig the oneview-sdk (5.0.0).

Also fixes all the consequences (mainly new rucobop style).

### Issues Resolved
#246 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass (`$ rake test`).
  - [ ] New resources and/or actions have are included in the [matchers.rb](https://github.com/HewlettPackard/oneview-chef/blob/master/libraries/matchers.rb) and [matchers_spec](https://github.com/HewlettPackard/oneview-chef/blob/master/spec/unit/resources/matchers_spec.rb).
- [ ] New functionality has been documented in the README if applicable.
  - [ ] New functionality has been thoroughly documented in [examples](https://github.com/HewlettPackard/oneview-chef/tree/master/examples/) (please include helpful comments).
- [ ] Changes documented in the [CHANGELOG](https://github.com/HewlettPackard/oneview-chef/blob/master/CHANGELOG.md).

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -2,16 +2,16 @@ source 'https://rubygems.org'
...
 
-gem 'berkshelf'
+gem 'berkshelf', '~> 4.3.5'
 gem 'chef', '~> 12.0'
-gem 'chefspec'
+gem 'chefspec', '~> 7.1'
 gem 'codeclimate-test-reporter', '~> 1.0.0'
-gem 'foodcritic'
-gem 'oneview-sdk', '~> 4.5.1'
+gem 'foodcritic', '~> 11.3'
```
Comment (fgbulsoni) [inline Gemfile]: This is causing an error in the CI for Ruby 2.1:
```
    foodcritic (~> 11.3) was resolved to 11.3.0, which depends on
      ruby (>= 2.2.2)
```
Should likely add it in a condition, similar to the guard gem, to use this version in Ruby 2.2+ and use the latest possible/available for Ruby 2.1.

**Review by fgbulsoni — CHANGES_REQUESTED** [review]

File: Gemfile (hunk 1 above)
Comment (tmiotto) [reply Gemfile]: Actually it is not the best approach.

I'll lower the version binding, it is not safe to leave it causing issues as new versions appears in the new rubies. (same as rubocop rules)

Comment (fgbulsoni) [reply Gemfile]: > I'll lower the version binding, it is not safe to leave it causing issues as new versions appears in the new rubies

Right, but that version is only causing issues in Ruby 2.1. 
Should we really stick to a lower version of that gem on all rubies, just for ruby 2.1 compatibility? 
Also, if it actually dropped ruby 2.1 support, that'd mean we'd be stuck on that version permanently.

Comment (tmiotto) [reply Gemfile]: > Should we really stick to a lower version of that gem on all rubies, just for ruby 2.1 compatibility?

Yes, the rules may change from one version to another, and we do not want to keep two different rule files and so on.

**Review by fgbulsoni — APPROVED** [review]

**Pull request merged**